src/main/java/views/LeagueSimulatorPane.java

```diff
@@ -0,0 +1,296 @@
+
+            centerContainer.getChildren().addAll(winnersText, nextRound);
+
+        }
+
+        centerContainer.getChildren().remove(centerContainer.getChildren().size() - 1);
+        currentLeagueId++;
+
+    }
+
+    void showError( String text ){
+        Text errorText = new Text(text);
+        errorText.setFill(Color.RED);
+        centerContainer.getChildren().add(errorText);
+    }
+
+    void refreshSelection() {
+        comboBox1.valueProperty().set(null);
+        comboBox2.valueProperty().set(null);
+        comboBox3.valueProperty().set(null);
+        comboBox4.valueProperty().set(null);
+        comboBox5.valueProperty().set(null);
+        comboBox6.valueProperty().set(null);
+        comboBox7.valueProperty().set(null);
+        comboBox8.valueProperty().set(null);
+        comboBox9.valueProperty().set(null);
+        comboBox10.valueProperty().set(null);
+        comboBox11.valueProperty().set(null);
+        comboBox12.valueProperty().set(null);
+        comboBox13.valueProperty().set(null);
+        comboBox14.valueProperty().set(null);
+        comboBox15.valueProperty().set(null);
+        comboBox16.valueProperty().set(null);
+    }
+
+    @Override
+    void setCallbacks() {
+        playButton.setOnMouseClicked(event -> {
+            centerContainer.getChildren().clear();
+            try {
+                this.createLeagueSchedule();
+                this.displayWinners();
+            } catch (Exception e) {
+                this.showError("Please select teams for all matches.");
+            }
+        });
+
+        returnButton.setOnMouseClicked(event -> SceneManager.getInstance().showPane((MainMenuPane.class)));
+        refreshButton.setOnMouseClicked(event -> refreshSelection());
+    }
+
+    void updateTeams() {
+        teamNames = GAME.getAllTeamNames();
+        comboBox1.setItems(FXCollections.observableList(teamNames));
+        comboBox2.setItems(FXCollections.observableList(teamNames));
+        comboBox3.setItems(FXCollections.observableList(teamNames));
+        comboBox4.setItems(FXCollections.observableList(teamNames));
+        comboBox5.setItems(FXCollections.observableList(teamNames));
+        comboBox6.setItems(FXCollections.observableList(teamNames));
+        comboBox7.setItems(FXCollections.observableList(teamNames));
+        comboBox8.setItems(FXCollections.observableList(teamNames));
+        comboBox9.setItems(FXCollections.observableList(teamNames));
+        comboBox10.setItems(FXCollections.observableList(teamNames));
+        comboBox11.setItems(FXCollections.observableList(teamNames));
+        comboBox12.setItems(FXCollections.observableList(teamNames));
+        comboBox13.setItems(FXCollections.observableList(teamNames));
+        comboBox14.setItems(FXCollections.observableList(teamNames));
+        comboBox15.setItems(FXCollections.observableList(teamNames));
+        comboBox16.setItems(FXCollections.observableList(teamNames));
+    }
+}
```

Comment:
A lot of similar lines being repeated again

Reply:
you could loop over comboBox with an iterator !